comp/core/workloadmeta/cached_entity.go

```diff
@@ -45,7 +45,35 @@ func (e *cachedEntity) set(source Source, entity Entity) (found, changed bool) {
 		return true, false
 	}
 
-	e.sources[source] = entity
```

Comment:
I'm not sure about changing the behavior of the merge here. For other collectors, we also face this problem. AFAIK we solved it by doing the merge directly inside the collector.